File: LeetCode/367有效的完全平方数.java
class Solution {
    public boolean isPerfectSquare(int num) {
        int left = 0, right = num;
        while (left <= right) {
            int mid = left + (right - left) / 2;
            long temp = (long)mid * mid;
            if (temp == num) return true;
            else if (temp < num) {
                left = mid + 1;
            } else {
                right = mid - 1;
            }
        } 
        return false;
    }
}

File: LeetCode/367有效的完全平方数.go
func isPerfectSquare(num int) bool {
	left, right := 0, num
	for left <= right {
		mid := left + (right-left)>>1
		if mid*mid == num {
			return true
		} else if mid*mid < num {
			left = mid + 1
		} else {
			right = mid - 1
		}
	}
	return false
}